src/main/java/net/smartcosmos/cluster/auth/SmartCosmosAuthenticationProvider.java

```diff
@@ -267,37 +232,13 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
             userResponse.getTenantUrn(),
             userResponse.getUserUrn(),
             userResponse.getUsername(),
-            getPasswordHash(userResponse),
+            userResponse.getPasswordHash(),
```

Comment:
Unless you changed something in `UserResponse` or the parentage of `SmartCosmosCachedUser` this won't work, because the final constructor in `User` will throw an exception if the password hash is `null` -- and it will be `null` in the refresh process, because the user details service can't return a password hash in that case. It simply doesn't have access to the password.